client/src/app/common/CustomRules/useRuleFiles.tsx

```diff
@@ -171,6 +171,9 @@ export default function useRuleFiles(
         )
     );
     setRuleFiles(updatedRuleFilesList);
+    if (!updatedRuleFilesList.some((file) => file.loadResult === "danger")) {
+      setUploadError("");
```

Comment:
This is a case where `uploadError` should probably not be its own state since it could be derived from the contents of `ruleFiles` at render time. But that's a more complex refactor, so we could revisit it and merge this for now.